internal/templating/templates_ingress.go

```diff
@@ -131,10 +132,17 @@ func GenerateIngressTemplate(
 		additionalAnnotations["nginx.ingress.kubernetes.io/ssl-redirect"] = "true"
 		additionalAnnotations["ingress.kubernetes.io/ssl-redirect"] = "true"
 	}
+
 	if lValues.EnvironmentType == "development" || route.Autogenerated {
 		additionalAnnotations["nginx.ingress.kubernetes.io/server-snippet"] = "add_header X-Robots-Tag \"noindex, nofollow\";\n"
 	}
 
+	if lValues.Oauth2ProxyDomain != "" && *route.OauthProtected {
+		// Protect all development and autogenerated routes with oauth2-proxy
+		additionalAnnotations["nginx.ingress.kubernetes.io/auth-signin"] = lValues.Oauth2ProxyDomain + "/oauth2/start?rd=$scheme://$host$request_uri"
+		additionalAnnotations["nginx.ingress.kubernetes.io/auth-url"] = lValues.Oauth2ProxyDomain + "/oauth2/auth?lagoon_environment=" + lValues.Project + "-" + lValues.Environment 
```

Comment:
Combining `Project`+`Environment` is not the right way to form a namespace, 90% of the time you might be able to get away with, however you will get stung by this. You can just consume the namespace though, and don't bother with `lagoon_environment`, since its just the namespace

```suggestion
		additionalAnnotations["nginx.ingress.kubernetes.io/auth-url"] = lValues.Oauth2ProxyDomain + "/oauth2/auth?namespace=" + lValues.Namespace
```

This is what [this](https://github.com/uselagoon/machinery/blob/8c481dfaf8b95e51df8b61ce8970ac5faa31fff6/utils/namespace/namespace.go#L60) does to generate the namespace

Reply:
changed.